File: 700-search-in-a-binary-search-tree/700-search-in-a-binary-search-tree.cpp
class Solution {
public:
TreeNode* searchBST(TreeNode* root, int val) {
if(root==NULL)
    return NULL;
    
if(val == root->val)
    return root;
    
else if(root->val>val)
    return searchBST(root->left, val);
    
else if(root->val<val)
    return searchBST(root->right, val);
    return NULL;

}
};

File: 700-search-in-a-binary-search-tree/700-search-in-a-binary-search-tree.cs
public class Solution {
    // Time O(n)
    public TreeNode SearchBST(TreeNode root, int val) {
        if(root == null) return root;
        if(root.val == val) return root;
        else if (root.val > val) return SearchBST(root.left,val);
        else return SearchBST(root.right,val);
    }
}
